### PR TITLE
fix(SyntheticReadAloud): associate derivatives with commits so that not every publish or commit needs a newly generated audio if not necessary

### PR DESCRIPTION
**Pull request by lucianak**
add feature to skip audio generation while publishing, and use audio from latest published commit instead. Adds new table to associate derivatives with commits.

### CODE REVIEW

File: packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts [hunk 1]
```diff
@@ -131,6 +140,32 @@ export const onPublish = async (document: any, pgdb: any, user?: any) => {
...
+        WHERE repos."currentPhase" in ('scheduled', 'published')
+        AND milestones.scope = 'publication'
+        AND d.status = 'Ready'
+        AND d.type = 'SyntheticReadAloud'
+        AND repos.id = :repoId
+        ORDER BY milestones."publishedAt" DESC;`,
+      { repoId: repoId },
+    )
+
+    await associateReadAloudDerivativeWithCommit(latestPublishedDerivative, commit, pgdb)
```
Comment (annatraussnig) [inline packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts:164]: do we handle the case where there is not previous derivative available?

**Review by hdahlheim — APPROVED** [review]
👍 Nothing to add

File: packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts [hunk 2]
```diff
@@ -164,22 +236,20 @@ export const derive = async (
...
-    //  (probably remove altogether when we delete the 'audio generieren' button)
-    if (pendingCount > 1) {
-      handlerDebug('too many pending derivatives. skipping synthesizing.', {
+    const pendingCount = await pgdb.queryOneField(`SELECT COUNT(d.id)
+      FROM publikator.derivatives d
+      JOIN publikator.commits c ON c.id = d."commitId"
+      WHERE status = 'Pending'
+      AND c."repoId" = :repoId;`, {repoId: document.repoId})
+      
+      console.log('pending count: %s', pendingCount)
```
Comment (annatraussnig) [inline packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts:245]: is this meant to stay?

**Review by annatraussnig — APPROVED** [review]

File: packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts (hunk 2 above)
Comment (lucianak) [reply packages/backend-modules/publikator/lib/Derivative/SyntheticReadAloud.ts:245]: oops

**Pull request merged**